spec/services/age_calculator_spec.rb

```diff
@@ -3,29 +3,51 @@
 RSpec.describe AgeCalculator do
   subject { described_class.call(date_of_birth, date) }
 
-  let(:date) { Date.parse("15-08-2010") }
+  let(:date_of_birth) { Date.parse("15-08-2000") }
 
-  context "birthday is just before date" do
-    let(:date_of_birth) { Date.parse("14-08-2000") }
+  context "when date is 1 day after birthday" do
+    let(:date) { Date.parse("16-08-2010") }
 
-    it("gives the right age") { expect(subject).to eq(10) }
+    it { is_expected.to eq(10) }
   end
 
-  context "birthday is just after date" do
-    let(:date_of_birth) { Date.parse("16-08-2000") }
+  context "when date is 1 day before birthday" do
+    let(:date) { Date.parse("14-08-2010") }
 
-    it("gives the right age") { expect(subject).to eq(9) }
+    it { is_expected.to eq(9) }
   end
 
-  context "birthday is same day as date" do
-    let(:date_of_birth) { Date.parse("16-08-2000") }
+  context "when date is same day as birthday" do
+    let(:date) { Date.parse("15-08-2010") }
 
-    it("gives the right age") { expect(subject).to eq(9) }
+    it { is_expected.to eq(10) }
   end
 
-  context "date_of_birth is on 29th of february and date is not a leap year" do
+  context "when birthday is on 29th of February" do
     let(:date_of_birth) { Date.parse("29-02-2000") }
 
-    it("gives the right age") { expect(subject).to eq(10) }
+    context "with date of 29th Ferbuary in a leap year" do
+      let(:date) { Date.parse("29-02-2012") }
+
+      it { is_expected.to eq(12) }
+    end
+
+    context "with date of 28th Ferbuary in a leap year" do
+      let(:date) { Date.parse("28-02-2012") }
+
+      it { is_expected.to eq(11) }
+    end
+
+    context "with date of 28th February in a non-leap year" do
+      let(:date) { Date.parse("28-02-2010") }
+
+      it { is_expected.to eq(9) }
+    end
+
+    context "with date of 1st March February in a non-leap year" do
```

Comment:
Should this be "with date of 1st March in a non-leap year"?

Reply:
thanks, done